sdk/src/main/java/com/qonversion/android/sdk/internal/mappers/UserMapper.kt

```diff
@@ -0,0 +1,41 @@
+package com.qonversion.android.sdk.internal.mappers
+
+import com.qonversion.android.sdk.internal.extensions.mapping.getString
+import com.qonversion.android.sdk.internal.extensions.mapping.getDate
+import com.qonversion.android.sdk.internal.extensions.mapping.getMap
+import com.qonversion.android.sdk.public.Entitlement
+import com.qonversion.android.sdk.public.User
+import com.qonversion.android.sdk.public.UserPurchase
+
+class UserMapper internal constructor(
+    private val purchasesMapper: Mapper<UserPurchase>,
+    private val entitlementMapper: Mapper<Entitlement>
+) : Mapper<User> {
+
+    override fun fromMap(data: Map<String, Any?>): User? {
+        val id = data.getString("id")
+
+        if (id.isNullOrEmpty()) {
+            return null
+        }
+
+        val createdDate = data.getDate("created")
+        val lastOnlineDate = data.getDate("last_online")
+
+        val entitlementsData = data.getMap("entitlements")
+        var entitlements: List<Entitlement> = emptyList()
+
+        val purchasesData = data.getMap("purchases")
+        var purchases: List<UserPurchase> = emptyList()
+
+        if (entitlementsData != null) {
```

Comment:
Is there is a way to move null check into the `arrayFromMap` func? 
`arrayFromMap` will return empty array for null value